File: programmers/lv0/코딩기초트레이닝/조건문과반복문/마지막두원소/Solution.swift
import Foundation

func solution(_ num_list:[Int]) -> [Int] {
    var answer = num_list
    let before = num_list[num_list.count - 2] // n-1
    let last = num_list[num_list.count - 1] // n

    if last > before {
        answer.append(last - before)
    } else {
        answer.append(last * 2)
    }

    return answer
}

File: programmers/lv0/코딩기초트레이닝/조건문과반복문/마지막두원소/Solution.go
package main

func solution(num_list []int) []int {
	var a int = num_list[len(num_list)-2:][0] // n - 1
	var b int = num_list[len(num_list)-2:][1] // n
	answer := make([]int, len(num_list)+1)
	for i, x := range num_list {
		answer[i] = x
	}
	if b > a {
		answer[len(answer)-1] = b - a
	} else {
		answer[len(answer)-1] = b * 2
	}
	return answer
}
func main() {
	solution([]int{2, 1, 6})
	solution([]int{5, 2, 1, 7, 5})
}
